modules/c++/sys/source/OSUnix.cpp

```diff
@@ -346,6 +346,53 @@ void sys::OSUnix::getMemInfo(size_t &totalPhysMem, size_t &freePhysMem) const
 #endif
 }
 
+std::string sys::OSUnix::getCurrentExecutable(
+        const std::string& argvPathname) const
+{
+    std::vector<std::string> possibleSymlinks;
+
+    // Linux
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self", "exe")));
+
+    // Solaris
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self",
+            sys::Path::joinPaths("path", "a.out"))));
+
+    for (size_t ii = 0; ii < possibleSymlinks.size(); ++ii)
+    {
+        struct stat linkStatus;
+        const std::string path = possibleSymlinks[ii];
+        if (lstat(path.c_str(), &linkStatus) == 0)
+        {
+            // Guessing a sane maximum based on Windows MAX_PATH
+            // If it's not enough, handle it through the base class
+            size_t bufferSize = 260 + 1;
+            char buffer[bufferSize];
```

Comment:
Can't put an array "on the stack" with a size set at runtime (even though here really this constant is known at compile time, it's not a static constant so it only sort of is).  gcc lets you get away with this (and likely does a dynamic allocation) but it's not portable.